metadata-ingestion/src/datahub/ingestion/source/sql/oauth_generator.py

```diff
@@ -0,0 +1,63 @@
+import base64
+import logging
+from typing import Any, Dict, List, Union
+
+import msal
+from OpenSSL.crypto import FILETYPE_PEM, load_certificate
+
+log = logging.getLogger(__name__)
+
+
+class OauthTokenGenerator:
+    def __init__(self, client_id, authority_url):
+        self.client_id = client_id
+        self.authority_url = authority_url
+
+    def _get_token(
+        self,
+        credentials: Union[str, Dict[str, Any]],
+        scope: Union[str, List[str]],
+        check_cache: bool,
+    ) -> str:
+        app = msal.ConfidentialClientApplication(
+            self.client_id, authority=self.authority_url, client_credential=credentials
+        )
+        token = None
+        if isinstance(scope, str):
+            scope = [scope]
+        if check_cache:
+            token = app.acquire_token_silent(scopes=scope, account=None)
+
+        if not token:
+            log.debug("No token available in cache. Getting a new one from AAD.")
+            token = app.acquire_token_for_client(scopes=scope)
+        return token
+
+    def get_public_certificate_thumbprint(self, public_cert_str: str) -> str:
+        cert_str = public_cert_str
+        certificate = load_certificate(FILETYPE_PEM, cert_str.encode("utf-8"))
+        return (certificate.digest("sha1")).decode("utf-8").replace(":", "")
+
+    def get_token_with_certificate(
+        self,
+        private_key_content: str,
+        public_key_content: str,
+        scope: Union[str, List[str]] = "https://management.core.windows.net/.default",
```

Comment:
This default seems very arbitrary. Please pass it to `OauthTokenGenerator` via the config option for Microsoft specific things.

Reply:
I guess we should be able to infer from the `host_name` whether this is Microsoft or not. We can add a warning to the source when `Oauth` is being used but snowflake is not on Microsoft.

Reply:
remove the deafult value